neuromation/cli/object_storage.py

```diff
@@ -0,0 +1,367 @@
+    else:
+        if root.tty:
+            formatter = VerticalColumnsFilesFormatter(
+                width=root.terminal_size[0], color=root.color
+            )
+        else:
+            formatter = SimpleFilesFormatter(root.color)
+
+    if len(obj_listings) > 1:
+        buffer = []
+        for uri, listing in zip(uris, obj_listings):
+            buffer.append(click.style(str(uri), bold=True) + ":")
+            buffer.extend(formatter(listing))
+        pager_maybe("".join(buffer), root.tty, root.terminal_size)
+    else:
+        assert obj_listings
+        pager_maybe(formatter(obj_listings[0]), root.tty, root.terminal_size)
+
+
+@command()
+@click.argument("patterns", nargs=-1, required=False)
+async def glob(root: Root, patterns: Sequence[str]) -> None:
+    """
+    List resources that match PATTERNS.
+    """
+    for pattern in patterns:
+        uri = parse_obj_resource(pattern, root)
+        if root.verbosity > 0:
+            painter = get_painter(root.color, quote=True)
+            curi = painter.paint(str(uri), FileStatusType.FILE)
+            click.echo(f"Using pattern {curi}:")
+        assert uri.host
+        if globmodule.has_magic(uri.host):
+            raise ValueError(
+                "You can not glob on bucket names. Please provide name explicitly."
+            )
+        objects = await root.client.obj.glob_objects(
+            bucket_name=uri.host, pattern=uri.path
+        )
+        for obj in objects:
+            click.echo(obj.uri)
+
+
+@command()
+@click.argument("sources", nargs=-1, required=False)
+@click.argument("destination", required=False)
+@option("-r", "--recursive", is_flag=True, help="Recursive copy, off by default")
+@option(
+    "--glob/--no-glob",
+    is_flag=True,
+    default=True,
+    show_default=True,
+    help="Expand glob patterns in SOURCES with explicit scheme.",
+)
+@option(
+    "-t",
+    "--target-directory",
+    metavar="DIRECTORY",
+    default=None,
+    help="Copy all SOURCES into DIRECTORY.",
+)
+@option(
+    "-T",
+    "--no-target-directory",
+    is_flag=True,
+    help="Treat DESTINATION as a normal file.",
+)
+@filter_option(
+    "--exclude",
+    "filters",
+    flag_value=True,
+    help=(
+        "Exclude files and directories that match the specified pattern. "
+        "The default can be changed using the storage.cp-exclude "
+        'configuration variable documented in "neuro help user-config"'
+    ),
+)
+@filter_option(
+    "--include",
+    "filters",
+    flag_value=False,
+    help=(
+        "Don't exclude files and directories that match the specified pattern. "
+        "The default can be changed using the storage.cp-exclude "
+        'configuration variable documented in "neuro help user-config"'
+    ),
+)
+@option(
+    "-p/-P",
+    "--progress/--no-progress",
+    is_flag=True,
+    default=True,
+    help="Show progress, on by default.",
+)
+async def cp(
```

Comment:
Removing functionality actually seemed harder for some cases, like not filtering was hard cause of how all helper functions reuse them. Only removed `update` functionality, as it caused some pain in development.